centreon/www/class/centreonAuth.LDAP.class.php

```diff
@@ -225,18 +225,32 @@ public function updateUserDn()
 
             //getting user's email
             $userEmail = $this->contactInfos['contact_email'];
-            if (
-                isset($userInfos[$this->ldap->getAttrName('user', 'email')])
-                && trim($userInfos[$this->ldap->getAttrName('user', 'email')]) != ''
-            ) {
-                if (is_array($userInfos[$this->ldap->getAttrName('user', 'email')])) {
-                    // Get the first if there are multiple entries
-                    if ($userInfos[$this->ldap->getAttrName('user', 'email')][0]) {
-                        $userEmail = $userInfos[$this->ldap->getAttrName('user', 'email')][0];
+            try {
+                if (isset($userInfos[$this->ldap->getAttrName('user', 'email')])) {
+                    if (is_array($userInfos[$this->ldap->getAttrName('user', 'email')])) {
+                        // Get the first if there are multiple entries
+                        if (
+                            $userInfos[$this->ldap->getAttrName('user', 'email')][0]
+                            && trim($userInfos[$this->ldap->getAttrName('user', 'email')][0]) !== ''
+                        ) {
+                            $userEmail = $userInfos[$this->ldap->getAttrName('user', 'email')][0];
+                        }
+                    } elseif (
+                        is_string($userInfos[$this->ldap->getAttrName('user', 'email')])
+                        && trim($userInfos[$this->ldap->getAttrName('user', 'email')]) !== ''
+                    ) {
+                        $userEmail = $userInfos[$this->ldap->getAttrName('user', 'email')];
+                    } else {
+                        throw new Exception('User email must be a string or an array of strings.');
```

Comment:
Simple question: according to the possible types value, wouldn't it be ok to have this instead (simpler to read)

```suggestion
                    $ldapEmailValue = $userInfos[$this->ldap->getAttrName('user', 'email')];
                    $userEmail = (trim(is_array($ldapEmailValue) ? current($ldapEmailValue): $ldapEmailValue));

                    if ($userEmail === '') {
                        throw new Exception('User email must be a string or an array of strings.');
```

Reply:
applied in separate commit